src/cmd/script_args/type_args.rs

```diff
@@ -0,0 +1,174 @@
+use clap::Parser;
+use move_core_types::language_storage::TypeTag;
+use move_core_types::parser::parse_type_tag;
+use std::{fmt, str::FromStr};
+
+/// TypeTag vector container.
+#[derive(Debug, Parser)]
+pub struct TypeArgVec {
+    /// TypeTag arguments separated by spaces.
+    ///
+    /// Example: `u8 u16 u32 u64 u128 u256 bool address vector signer`.
+    #[clap(long, multiple_values(true))]
+    pub(crate) type_args: Vec<MoveType>,
+}
+
+// This function cannot handle the full range of types that MoveType can
+// represent. Internally, it uses parse_type_tag, which cannot handle references
+// or generic type parameters. This function adds nominal support for references
+// on top of parse_type_tag, but it still does not work for generic type params.
+// For that, we have the Unparsable variant of MoveType, so the deserialization
+// doesn't fail when dealing with these values.
+impl FromStr for MoveType {
+    type Err = anyhow::Error;
+
+    fn from_str(mut s: &str) -> Result<Self, Self::Err> {
```

Comment:
Why is it `mut s: &str`? :upside_down_face:

Reply:
_True reason: this code has been purely taken from the `aptos-core/api/types/src/move_types.rs` file._ 🙃
 
But it does seem ugly - let's fix it by doing variable shadowing in the next line. (since we do mutate the `s` in the function)
Done